frontend/src/pages/event.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react'
+import { Table, Row, Button, Col, FormGroup, Label, Input, Container } from 'reactstrap'
+import ActionButton from '../components/actionButton'
+import Nav from '../components/nav'
+import Head from '../components/head'
+import { getEventById, getEventAttendees } from '../utils/api'
+import Router from 'next/router'
+import Link from 'next/link'
+
+class Event extends React.Component<Props> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      name: '',
+      date: '',
+      startTime: '',
+      endTime: '',
+      location: '',
+      description: '',
+      fbLink: '',
+      attendeeEmails: [],
+      attendees: []
+    }
+  }
+
+  async componentDidMount() {
+    const { query } = Router
+    const { result } = await getEventById(query.id)
+    this.setState({
+      name: result.name,
```

Comment:
this could break if the request fails

Reply:
could probably just say
```js
result && this.setState({...})
```

Reply:
Double check on this before you fix alice. It might just assign everything to undefined, since I believe failing requests still return a result (with status code and error message, etc.). Not sure tho, maybe it would break.

Reply:
it will break

Reply:
why we do this https://github.com/hack4impact-uiuc/h4i-recruitment/blob/ea5ad18eb4e637ca940228cc540ca228b0c7a022/frontend/src/pages/candidate.js#L53